hack/generate/prometheus-operator.jsonnet

```diff
@@ -15,6 +15,8 @@ local operatorContainer =
   container.new("prometheus-operator", "quay.io/coreos/prometheus-operator:v" + version) +
   container.withPorts(containerPort.newNamed("http", targetPort)) +
   container.withArgs(["--kubelet-service=kube-system/kubelet", "--config-reloader-image=quay.io/coreos/configmap-reload:v0.0.1"]) +
+  container.mixin.securityContext.withAllowPrivilegeEscalation(false) +
+  container.mixin.securityContext.withReadOnlyRootFilesystem(true) +
```

Comment:
Does this collide with PSPs? Generally I'm fine with it, but if this causes the same troubles as specifying the Pod's security context, then I'd rather hold off.

Reply:
I am not quite sure I understand your comment. Could you elaborate further on this?

`AllowPrivilegeEscalation ` and `ReadOnlyRootFilesystem ` are only options in the container security context, not in the pod security context.

Reply:
Maybe the description of the `SecurityContext` struct helps:

```golang
// SecurityContext holds security configuration that will be applied to a container.
// Some fields are present in both SecurityContext and PodSecurityContext.  When both
// are set, the values in SecurityContext take precedence.
type SecurityContext struct {
```

Reply:
Yes, I'm referring to the precedence taken here. If a PodSecurityPolicy sets something but the container's SecurityContext specifies something different, then does that prevent the Pod from being scheduled? Also are those not the default settings?

Reply:
Default of [`AllowPrivilegeEscalation`](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation) and [`ReadOnlyRootFilesystem`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#securitycontext-v1-core)  is `false` (see links).

Reply:
Default of [`AllowPrivilegeEscalation`](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privilege-escalation) is allow. Default of [`ReadOnlyRootFilesystem`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#securitycontext-v1-core)  is `false`.